src/DotNetty.Buffers/ReadOnlyByteBufferStream.cs

```diff
@@ -23,24 +23,26 @@ public ReadOnlyByteBufferStream(IByteBuffer buffer, bool releaseReferenceOnClosu
 
         public override bool CanRead => true;
 
-        public override bool CanSeek => false;
+        public override bool CanSeek => true;
 
         public override bool CanWrite => false;
 
         public override long Length
         {
-            get { throw new NotSupportedException(); }
+            get => this.buffer.ReaderIndex + this.buffer.ReadableBytes;
         }
 
         public override long Position
         {
-            get { throw new NotSupportedException(); }
-            set { throw new NotSupportedException(); }
+            get => this.buffer.ReaderIndex;
+            set => this.buffer.SetReaderIndex((int)value);
         }
 
         public override int Read(byte[] output, int offset, int count)
         {
-            int read = Math.Min(count - offset, this.buffer.ReadableBytes);
+            if (offset + count > output.Length) throw new ArgumentException();
+            int sourceBytes = Math.Max(0, this.buffer.ReadableBytes - this.buffer.ReaderIndex);
+            int read = Math.Min(count, sourceBytes);
```

Comment:
Yep you're right, thanks for catching this. I missed a unit test that would've caught this dumb error; adding it...